### PR TITLE
Index Changes

### PR DESCRIPTION
**Pull request by fbane**
If this does not pass QA lol.. I think I shall cry slowly into my pillow :( And then pet my dog.

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -101,16 +101,27 @@ function mysort($a, $b)
...
-                                '<tr>
-                                <!-- Owner of Map -->
-                                <td><a href="'.WWW_ROOT.'maps/' . $file. '">' . substr($file, 0, -5) . '</a></td>
+                            // Check to see if savefile exists in public folder and create download link
+                            if (file_exists(SHELL_ROOT.'saves/public/' . $saveFile)) {
 
+                              $uploaded = filemtime(SHELL_ROOT.'saves/public/'.$saveFile); // Checks the timestamp of saves in the public folder
+                              $timediff = time() - $uploaded; // Measure difference between current time and save file creation time
+                              $expired = round($timediff/1000/60,1); // Formula to show how many days and round to 2 decimal places
+
```
Comment (ian76g) [inline index.php:112]: wtf?
times are measured in seconds.
60 * 60 * 24 * 2 (seconds * minutes * hours * days) = 172800 (two days) - please remove this line

File: index.php [hunk 2]
```diff
@@ -101,16 +101,27 @@ function mysort($a, $b)
...
+                                // Timecheck to remove public link for download
+                                if ($expired < 1.45) {
+                                    echo '<td><a href="'.WWW_ROOT.'maps/' .$file. '">'.substr($file, 0, -5) .'</a> <a href="'.WWW_ROOT.'saves/public/'.$saveFile.'">(DL)</a></td>';
```
Comment (ian76g) [inline index.php:115]: if ($timediff < (60 * 60 * 24 * 2)) {    // two days

File: index.php [hunk 3]
```diff
@@ -120,29 +131,15 @@ function mysort($a, $b)
...
-                                <td>' . round($db[substr($file, 0, -5) . '.sav'][4]) . '%</td>
-                                <!-- Shared Link -->';
-
-                            // Checks public save folder to see if we can provide a link
-                            $saveCheck = SHELL_ROOT . 'saves/public/' . substr($file, 5, -5) . '.sav';
-                            if (file_exists($saveCheck)) {
-                                $upTime = filemtime($saveCheck);
-                                $timeCheck = time() - $upTime;
-                                if ($timeCheck < 172800) {
-                                    echo '<td><a href="' . WWW_ROOT . 'saves/public/' . substr($file, 5, -5) . '.sav">Link</a></td>';
```
Comment (ian76g) [inline index.php]: uh... here we had this magic number... ;-)

Comment (fbane) [reply index.php]: Sorry not sure what you mean by magic number?

**Review by fbane — COMMENTED** [review]
Wasn't sure what you meant by that magic number but others were updated.

I'm still learning about timestamps in php :\

**Review by ian76g — CHANGES_REQUESTED** [review]
whils this seems to might work - may I ask the question:

where is the code change, that puts public saves to saves/public instead of /public?
Seems this is only the frontend part - but in this PR I request also to change the upload part, that still would put public saves to the "wrong" folder ;-)

**Review by ian76g — APPROVED** [review]

**Pull request merged**